subsys/net/lib/sockets/sockets_misc.c

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <net/socket.h>
+
+int zsock_gethostname(char *buf, size_t len)
+{
+	const char *p = net_hostname_get();
+
+	strncpy(buf, p, len);
```

Comment:
strncpy() will fail to NUL-terminate buf if there's no space in the buffer for it, and gethostname() always writes a NUL-terminated hostname there